ostd/src/mm/vm_space.rs

```diff
@@ -433,6 +386,44 @@ impl CursorMut<'_, '_> {
         self.dispatch_tlb_flush();
     }
 
+    /// Copies the mapping from the given cursor to the current cursor.
+    ///
+    /// All the mappings in the current cursor's range must be empty. The
+    /// function allows the source cursor to operate on the mapping before
+    /// the copy happens. So it is equivalent to protect then duplicate.
+    /// Only the mapping is copied, the mapped pages are not copied.
+    ///
+    /// After the operation, both cursors will advance by the specified length.
+    ///
+    /// # Panics
+    ///
+    /// This function will panic if:
+    ///  - either one of the range to be copied is out of the range where any
+    ///    of the cursor is required to operate;
+    ///  - either one of the specified virtual address ranges only covers a
+    ///    part of a page.
+    ///  - the current cursor's range contains mapped pages.
+    pub fn copy_from(
+        &mut self,
+        src: &mut Self,
+        len: usize,
+        op: &mut impl FnMut(&mut PageProperty),
+    ) {
+        let va = src.virt_addr();
+
+        // SAFETY: Operations on user memory spaces are safe if it doesn't
+        // involve dropping any pages.
+        unsafe { self.pt_cursor.copy_from(&mut src.pt_cursor, len, op) };
+
+        if len > TLB_FLUSH_ALL_THRESHOLD * PAGE_SIZE {
+            src.issue_tlb_flush(TlbFlushOp::All, None);
+        } else {
+            src.issue_tlb_flush(TlbFlushOp::Range(va..va + len), None);
+        }
```

Comment:
This is one of the correct approaches, but currently, if there are many mappings and each mapping is not large, wouldn't this lead to a higher number of TLB flush operations during fork? I think it would be more reasonable to perform a unified flush after the fork is completed.